markdown/events/2023/bytesize_precommit.md

```diff
@@ -23,3 +23,39 @@ It is our hope that these talks / videos will build an archive of training mater
 ## Bytesize: nf-core/pre-commit
 
 This week, Matthias Hörtenhuber ([@mashehu](https://github.com/mashehu)) is going to explain the use the newly added pre-commit tool added to nf-core/tools. Pre-commits were developed to inspect the snapshot that is about to be committed and helps to check formatting etc. before adding the code to the repository.
+
+<details markdown="1"><summary>Video transcription</summary>
+**Note: The content has been edited for reader-friendliness**
+
+[0:01](https://www.youtube.com/watch?v=08d6zv6zvdM&t=1)
+(host) Hello, everyone, to today's Bitesize talk. I'm Franziska Bonath. I'm the host today. With me is Matthias Hörtenhuber. He is going to talk about pre-commit.
+
+[0:11](https://www.youtube.com/watch?v=08d6zv6zvdM&t=11)
+Yes. Hi. It will be a short one. Let's get going with it. The title is "pre-commit, hooked on a commit". It's about how we're trying to solve this problem a lot of us run into: You commit your code, then GitHub CI runs prettier, it finds a mistake and asks very passive aggressively, if you maybe forgot to run prettier. One of the solutions. First solution is, of course, you ask nf-core-bot to please fix the linting. I did a quick search. Actually, 260 PRs have a comment, at least one comment, where somebody asked the bot to fix the linting. That's nice that people are using the bot. But actually, maybe we should try to save the icebears and give nf-core-bot sometimes a break, because these things we can do differently.
+What if you don't need to run the bot, because you already run prettier when you commit your code, or actually before you commit your code? We use this tool called pre-commit, which runs prettier when you hit `git commit`. It runs prettier, checks if the code is fine and makes changes on it.
+
+[1:43](https://www.youtube.com/watch?v=08d6zv6zvdM&t=103)
+How do you set pre-commit up? Well, the good news is it comes already pre-installed with the nf-core tools as a dependency. Then you need to have a pre-commit config .yaml file. For example, this one here for prettier. Also, more good news, in the next tools release, this will be part of every pipeline template. Also in the modules repository. All there, we have that set up. It doesn't change anything for you that we have it there. But if you then also run in your repository `pre-commit install`, it actually sets up this git hook. Whenever you hit commit, prettier is run beforehand and doesn't allow you to commit until you fixed these changes.
```

Comment:
```suggestion
How do you set pre-commit up? Well, the good news is it comes already pre-installed with the nf-core tools as a dependency. Then you need to have a `pre-commit-config.yaml` file. For example, this one here for prettier. Also, more good news, in the next tools release, this will be part of every pipeline template. Also in the modules repository, we have that set up. It doesn't change anything for you that we have it there. But if you then also run in your repository `pre-commit install`, it actually sets up this git hook. Whenever you hit commit, prettier is run beforehand and doesn't allow you to commit until you fixed these changes.
```